doc/design/dist/submit-job.md

```diff
@@ -0,0 +1,161 @@
+
+- `Runtime docker image` and `base docker image`
+
+  For a distributed training job, there is two docker image called `runtime docker image` and `base docker image`, the `runtime docker image` is actually running in kubernetes.
+
+  - `runtime docker image` include user's package files and all dependencies.
+  - `base docker image` usually is PaddlePaddle product docker image including paddle binary files and some scripts used for starting up the trainer process and fetch some information of pod. And of course, users can also build their own paddle binary files into the custom `base docker image` with [this doc](../../../paddle/scripts/docker/README.md).
+
+  `runtime docker image` will be built by PaddlePaddle client automatic, here is a simple example project:
+  ```bash
+  paddle_example
+    |-quick_start
+      |-trainer.py
+      `-dataset.py
+    requirments.txt
+  ```
+  - `requirements.txt` list python dependencies package, you can create is like:
+
+    ```txt
+    pillow
+    protobuf==3.1.0
+    ```
+    some other details is [here](https://pip.readthedocs.io/en/1.1/requirements.html).
+  - `quick_start` directory include the trainer package files.
+
+  Execute the command: `paddle train...`, PaddlePaddle client will upload the trainer package files and setup parameters to [Paddle Server](#paddle-server).
+
+## Paddle Server
+Paddle server is running on kubernetes, users will configure the server address in [PaddlePaddle client configuration file](#configurate-paddlepaddle-client)
+
+- HTTP server
+
+  Paddle server is an HTTP server, Receiver the trainer files and saves them on GlustereFS.
+
+- Startup pservers and trainers
+
+  Paddle Service will deploy pserver and trainer on kubernetes, they are also job resource, naming `<job-name>-pserver` and `<job-name>-trainer`.
+  - Setup pserver job.
+  - Setup trainer job, trainer process will be setup until the status of all pserver pod becoming `RUNNING` and fetch all pserver's IP as trainer setup parameters.
+  - Setup trainer process in trainer pod.
+
+## Data source
+- Distributed file system
+
+  You can upload your training data to distributed file system, such as GlustereFS,
+  PaddlePaddle support a default reader for reading data from distributed file system.
+- HTTP server
+
+  TODO
+- Real-time data
+
+  TODO
+
+## PaddlePaddle client commands:
+To run local training job with flag `--locally` and distributed training job without it.
+- `paddle train`: start a training job
+- `paddle prediction`: start a prediction job
+- `paddle list`: list all PaddlePaddle jobs in current namespace
+- `paddle cancel`: cancel a running job.
+- `paddle status`: status of a PaddlePaddle job
+- `paddle version`: show PaddlePaddle client and PaddlePaddle server version info.
+
+## Work feature
+- V1
+  - Paddle server is a local version, build `runtime docker image`, deploy trainer and pserver job on user's host.
+  - implement `paddle train`, `paddle list`, `paddle cancel`
```

Comment:
If we let user do `python train.py` for running on local, we can change `paddle train` to something like `paddle submit`.